aws_batch/postprocess.py

```diff
@@ -0,0 +1,244 @@
+    accession2taxid_dict = shelve.open(accession2taxid_path)
+    lineage_map = shelve.open(lineagePath)
+    input_fasta_f = open(input_fasta_file, 'rb')
+    output_fasta_f = open(output_fasta_file, 'wb')
+    sequence_name = input_fasta_f.readline()
+    sequence_data = input_fasta_f.readline()
+    while len(sequence_name) > 0 and len(sequence_data) > 0:
+        read_id = sequence_name.rstrip().lstrip('>') # example read_id: "NR::NT:CP010376.2:NB501961:14:HM7TLBGX2:1:23109:12720:8743/2"
+        nr_taxid_species, nr_taxid_genus, nr_taxid_family = accession2taxid(read_id, accession2taxid_dict, 'NR', lineage_map)
+        nt_taxid_species, nt_taxid_genus, nt_taxid_family = accession2taxid(read_id, accession2taxid_dict, 'NT', lineage_map)
+        new_read_name = ('nr:' + nr_taxid_species + ':nt:' + nt_taxid_species
+                         + ':' + read_id)
+        output_fasta_f.write(">%s\n" % new_read_name)
+        output_fasta_f.write(sequence_data)
+        sequence_name = input_fasta_f.readline()
+        sequence_data = input_fasta_f.readline()
+    input_fasta_f.close()
+    output_fasta_f.close()
+
+def get_taxid(sequence_name, taxid_field):
+    parts = sequence_name.split(":")
+    if len(parts) < taxid_field:
+         return 'none'
+    return parts[taxid_field-1]
+
+def generate_taxid_locator(input_fasta, taxid_field, output_fasta, output_json):
+    # put every 2-line fasta record on a single line with delimiter ":lineseparator:":
+    command = "awk 'NR % 2 == 1 { o=$0 ; next } { print o \":lineseparator:\" $0 }' " + input_fasta
+    # sort the records based on the field containing the taxids:
+    command += " | sort --key %s --field-separator ':' --numeric-sort" % taxid_field
+    # split every record back over 2 lines:
+    command += " | sed 's/:lineseparator:/\\n/g' > %s" % output_fasta
+    subprocess.check_output(command, shell=True)
+    # make json giving byte range of file corresponding to each taxid:
+    taxon_sequence_locations = []
+    f = open(output_fasta, 'rb')
+    sequence_name = f.readline()
+    sequence_data = f.readline()
+    taxid = get_taxid(sequence_name, taxid_field)
+    first_byte = 0
+    new_first_byte = first_byte + len(sequence_name) + len(sequence_data)
+    while len(sequence_name) > 0 and len(sequence_data) > 0:
+        sequence_name = f.readline()
+        sequence_data = f.readline()
+        new_taxid = get_taxid(sequence_name, taxid_field)
+        if new_taxid != taxid:
+            taxon_sequence_locations.append({'taxid': taxid, 'first_byte': first_byte,
+                                             'last_byte': new_first_byte - 1})
+            taxid = new_taxid
+            first_byte = new_first_byte
+            new_first_byte = first_byte + len(sequence_name) + len(sequence_data)
+        else:
+            new_first_byte += len(sequence_name) + len(sequence_data)
+    f.close()
+    with open(output_json, 'wb') as f:
+       json.dump(taxon_sequence_locations, f)
```

Comment:
How big do these files tend to be again?

Reply:
10-100 KB